packages/liveblocks-react/src/umbrella-store.ts

```diff
@@ -1025,6 +1033,57 @@ export class UmbrellaStore<M extends BaseMetadata> {
     this.setQuery2State(queryKey, { isLoading: false, error });
   }
 
+  public setLoadThreadsPromise(
+    queryKey: string,
+    getThreadsAndNotificationsPromise: Promise<{
+      threads: ThreadData<M>[];
+      inboxNotifications: InboxNotificationData[];
+    }>
+  ): Promise<ThreadData<M>[]> {
```

Comment:
The goal of this was to delegate as much work as I could to the store, including auto retrying. There was still some work that I just couldn't pull out of the `extras` unless I spent a lot of time refactoring a lot of the code.